cmd/user-manager/manager/manager.go

```diff
@@ -0,0 +1,120 @@
+package manager
+
+import (
+	"database/sql"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/percona/percona-xtradb-cluster-operator/cmd/user-manager/db"
+	"github.com/pkg/errors"
+)
+
+type Manager struct {
+	db         *sql.DB
+	secretPath string
+	Users      []User
+}
+
+type UsersSecret struct {
+	StringData map[string]string `yaml:"stringData"`
+}
+
+type Data struct {
+	Users []User `yaml:"users"`
+}
+
+type User struct {
+	Name   string   `yaml:"username"`
+	Pass   string   `yaml:"password"`
+	Tables []Table  `yaml:"tables"`
+	Hosts  []string `yaml:"hosts"`
+}
+
+type Table struct {
+	Name       string `yaml:"name"`
+	Privileges string `yaml:"privileges"`
+}
+
+func New(hosts []string, rootPass string) (Manager, error) {
+	var um Manager
+	var err error
+	for _, host := range hosts {
+		mysqlDB, err := db.Conn("root:" + rootPass + "@tcp(" + host + ")/")
+		if err != nil {
+			return um, errors.Wrap(err, "create db connection")
+		}
+		um.db = mysqlDB
+		log.Println("using  host: " + host)
+		break
+	}
+	if um.db == nil {
+		return um, errors.Wrap(err, "cannot connect to any host")
+	}
+	um.secretPath = "./secret.yaml"
+
+	return um, nil
+}
+
+func (u *Manager) GetUsers() error {
+	secret, err := u.readSecretFile()
+	if err != nil {
+		return errors.Wrap(err, "read secret")
+	}
+	var data Data
+	err = yaml.Unmarshal(secret, &data)
+	if err != nil {
+		return errors.Wrap(err, "unmarshal secret")
+	}
+	u.Users = data.Users
+
+	return nil
+}
+
+func (u *Manager) readSecretFile() ([]byte, error) {
+	file, err := os.Open(u.secretPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "open secret file")
+	}
+	defer file.Close()
+	b, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, errors.Wrap(err, "read secret file")
+	}
+
+	return b, nil
+}
+
+func (u *Manager) ManageUsers() error {
```

Comment:
I believe all these user stuff should be done in a transaction. Otherwise, if some error occurs during the user creation we could end up in a situation with no users at all.